pkg/gpu/nvidia/manager.go

```diff
@@ -235,6 +244,106 @@ func (ngm *nvidiaGPUManager) CheckDeviceStates() bool {
 	return changed
 }
 
+// CheckHealth registers event with NVML to detect GPU critical errors
+func (ngm *nvidiaGPUManager) CheckHealth() error {
```

Comment:
To prevent the manager package from bloating too much, would it be better to move the health check functionality to a separate package? The device manager and health checker just need a channel to communicate. Separating health check piece out would make it easier to unit test, and read/make changes in the future.

Reply:
I was thinking about this as well. But CheckHealth basically deals with the device status, so logically it should be part of device manager.
If we separate it out, we will need one channel between `CheckHealth` and device manager so that device manager can update the status, and another channel between device manager and `ListAndWatch`. And that seems a bit too complicated. Thoughts?